### PR TITLE
fix(auth): Added missing UserContextData to Cognito User Pool operation SignUp

### PR DESCRIPTION
**Pull request by armandsyah134**
#### Description of changes

- Adding missing property `UserContextData` in `SignUp` call to Cognito User Pools (CUP)
- `UserContextData` is an important property for Sign Up calls, and should be added to the `SignUp` operation
    - Other CUP apis are passing in `UserContextData` (e.g. [ConfirmSignUp](https://github.com/aws-amplify/amplify-js/blob/main/packages/auth/src/providers/cognito/apis/confirmSignUp.ts#L72))
- `UserContextData` is already in your API [contract](https://github.com/aws-amplify/amplify-js/blob/main/packages/auth/src/providers/cognito/utils/clients/CognitoIdentityProvider/types.ts#L1477) 

#### Issue #, if available

N/a (will make an issue for this if required

#### Description of how you validated changes

- Added unit test for validating `UserContextData` present in SignUp call
- Performed local testing by linking changed library to sample NextJS app and testing sign up flow
    - Validated that UserContextData was present in our logs

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [n/a] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by israx — DISMISSED** [review]
LGTM

File: yarn.lock
Comment (jimblanc) [inline yarn.lock:1]: We can omit the yarn.lock change from this PR

Comment (armandsyah134) [reply yarn.lock:1]: omitted yarn lock file from pr

**Review by israx — APPROVED** [review]

**Review by jimblanc — CHANGES_REQUESTED** [review]
Change looks good to me, but please hold off on merging until we can discuss (blocking PR). Thank you for the contribution!

File: packages/auth/src/providers/cognito/apis/signUp.ts [hunk 1]
```diff
@@ -71,9 +71,18 @@ export async function signUp(input: SignUpInput): Promise<SignUpOutput> {
...
 		setAutoSignInStarted(true);
 	}
+
+	const { userPoolId, userPoolClientId } = authConfig;
+
+	const UserContextData = getUserContextData({
+		username,
+		userPoolId,
+		userPoolClientId,
+	});
```
Comment (cshfang) [inline packages/auth/src/providers/cognito/apis/signUp.ts:81]: Nit: Can we move this directly into line 96? I think `getUserContextData` is expressive enough to convey the context there when used directly. Also prefer not to have the PascalCased variable as it easily gets confused with types/interfaces/classes

**Review by jimblanc — APPROVED** [review]

**Review by haverchuck — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Review by israx — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**